client/js/components/App.react.jsx

```diff
@@ -16,24 +18,51 @@ export default class App extends React.Component {
     constructor(props) {
         super(props);
         this.state = {park: null,
-                      parkList: []};
-        getParksJson().then((data) => this.setState({parkList: data}));
+                      parkList: [],
+                      parkByID: {}};
+        getParksJson().then((data) => {
+            var parkByID = {}
+            data.forEach((current, index) => {
+                // NOTE: Coordinates are lng lat here
+                current.center.reverse();
+                parkByID[current.park_id] = current;
+            });
+            this.setState({parkList: data, parkByID: parkByID});
+        });
+    }
+
+    selectPark(parkID) {
+        return $.when(
+            getFeatureGeoJson(parkID, "park"),
+            getFeatureGeoJson(parkID, "amenity"),
+            getFeatureGeoJson(parkID, "facility"),
+            getFeatureGeoJson(parkID, "trail")).done(
+            (park, amenity, facility, trail) => {
+                this.setState({
+                    parkID: parkID,
+                    park: park,
+                    trail: trail,
+                    facility: facility,
+                    amenity: amenity,
+                    center: this.state.parkByID[parkID].center
+                });
+            }
+        );
     }
 
     render() {
-        var parkList = this.state.parkList.map((park) => {
-            return <li>{park.name}</li>;
-        });
-        return <ul>{parkList}</ul>;
+        var s = this.state;
+        //FIXME: Ain't got no CSS
```

Comment:
Readable code is readable code. I'm calling kneetree out for not including a space after the `//` symbol.

He did:

```js
//FIXME: Ain't
```

He should've done

```js
// FIXME: Ain't
```